src/lib/RichText/TextExtractor/FullTextExtractor.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\FieldTypeRichText\RichText\TextExtractor;
+
+use DOMDocument;
+use DOMNode;
+use Ibexa\Contracts\FieldTypeRichText\RichText\TextExtractorInterface;
+
+/**
+ * Extracts text content of the given $document.
+ *
+ * @internal Only for use by RichText FieldType itself.
+ */
+final class FullTextExtractor implements TextExtractorInterface
+{
+    public function extractText(DOMDocument $document): string
+    {
+        return $this->extractTextFromNode($document->documentElement);
```

Comment:
While at it, could we make it more bulletproof?
```suggestion
        return return null !== $document->documentElement ? $this->extractTextFromNode($document->documentElement) : '';
```
This should get rid of PHPStan issue as well.